Src/Particle/AMReX_ParticleCommunication.H

```diff
@@ -571,8 +571,8 @@ void communicateParticlesStart (const PC& pc, ParticleCopyPlan& plan, const SndB
         AMREX_ASSERT(plan.m_snd_counts[i] % ParallelDescriptor::sizeof_selected_comm_data_type(plan.m_snd_num_particles[i]*psize) == 0);
         AMREX_ASSERT(Who >= 0 && Who < NProcs);
 
-        ParallelDescriptor::Send((char const*)(snd_buffer.dataPtr()+snd_offset), Cnt, Who, SeqNum,
-                                 ParallelContext::CommunicatorSub());
+        ParallelDescriptor::Asend((char const*)(snd_buffer.dataPtr()+snd_offset), Cnt, Who, SeqNum,
```

Comment:
This will lead to a memory leak, because MPI_Request is not "freed" with MPI_Wait. We had this kind of leaks before, and it caused issues for long runs. So we know this does leak a small amount of memory in practice, not just in theory. See also https://stackoverflow.com/questions/43004420/is-a-mpi-wait-necessary-after-a-mpi-isend.

Since we use C++17 now. I suggest we make ParallelDescriptor::Asend `[[nodiscard]]`.